ml-proto/src/arithmetic.ml

```diff
@@ -50,7 +52,14 @@ struct
       | Abs -> Int.abs
       | Not -> Int.lognot
       | Clz -> fun i -> i  (* TODO *)
-      | Ctz -> fun i -> i  (* TODO *)
+      | Ctz -> fun i ->
+        if Int.to_int i == 0
```

Comment:
`==` is physical equality in OCaml, you want `=`. Also, could say `i = Int.zero` instead, which is shorter.